### PR TITLE
Adding property for setting the pulse width directly

### PR DESCRIPTION
**Pull request by kelliott121**
PR to accomplish #41

### CODE REVIEW

**Review by theacodes — APPROVED** [review]

File: adafruit_motor/servo.py [hunk 1]
```diff
@@ -71,6 +71,25 @@ def fraction(self, value):
...
         self._pwm_out.duty_cycle = duty_cycle
 
+    @property
+    def pulse_width(self):
+        """The pulse width sent to the servo in microseconds. Must be in
+        the range ``min_pulse`` to ``max_pulse``. Is None when servo is disabled.
+        """
+        if self._pwm_out.duty_cycle == 0:  # Special case for disabled servos
+            return None
+        return self._pwm_out.duty_cycle
```
Comment (tannewt) [inline adafruit_motor/servo.py:81]: This is incorrect because it's not doing the reverse math of line 88.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thank you for this PR! Unfortunately, I'm wary of including it because it'll make the library a little big larger. This library is frozen into the CircuitPlayground Crickit build which is basically out of space.

Could you add a calibration example instead that directly sets duty_cycle based on a target pulse length? Thanks!

File: adafruit_motor/servo.py (hunk 1 above)
Comment (kelliott121) [reply adafruit_motor/servo.py:81]: Whoops, thanks for the catch!

**Comment by kelliott121** [comment]
> Thank you for this PR! Unfortunately, I'm wary of including it because it'll make the library a little big larger. This library is frozen into the CircuitPlayground Crickit build which is basically out of space.
> 
> Could you add a calibration example instead that directly sets duty_cycle based on a target pulse length? Thanks!

Sure thing! I'm new to CircuitPython and ended up here from Adafruit Learn. Is there a better place to add features to servo motors?

**Comment by tannewt** [comment]
@kelliott121 Normally this would be fine but this library is very memory sensitive. You can either create another library with more features or just a "calibration" example that includes the appropriate math.

**Comment by kattni** [comment]
@kelliott121 Thank you for the contribution! Are you still available to work on this?

**Comment by kelliott121** [comment]
> @kelliott121 Thank you for the contribution! Are you still available to work on this?

Yes, sorry, got sidetracked on another project. I'll set a remind and can write a calibration example this weekend!

**Comment by jposada202020** [comment]
@kelliott121 Hello, hope you are good and well, any news on this? let us know if you are still interested on working on this. Thanks

**Comment by FoamyGuy** [comment]
@kattni this could get closed for now maybe. If original contributor does get back to it they can re-open or if someone else wants to take it on we could have a new PR for it.

**Pull request closed without merging**